multiversx_sdk/smart_contracts/smart_contract_controller.py

```diff
@@ -88,6 +88,7 @@ def create_transaction_for_deploy(
         transaction.nonce = nonce
 
         self._set_version_and_options_for_hash_signing(sender, transaction)
+        self._add_extra_gas_limit_if_required(transaction)
```

Comment:
I wonder whether we should also add a flag in entrypoint config - to be passed to controllers, to instruct them whether we should apply the gas limit adjustments or not. In some cases, maybe developers don't want that.